packages/core-components/src/layout/Sidebar/Page.tsx

```diff
@@ -163,3 +174,13 @@ export function useContent() {
 
   return { focusContent, contentRef: content?.contentRef };
 }
+
+/**
+ * Hook to read and update context details of a page with SideBar.
+ *
+ * @public
+ */
+export function usePageContext(): PageContextType {
+  const pageContext = useContext(PageContext);
+  return pageContext;
+}
```

Comment:
Sorry we need to find a cleaner way to achieve this, it's a very leaky abstraction

Reply:
Originally, I had implemented a new context for the page-metadata but then found this one, so I tried to reuse it.

I was thinking about introducing sort of `usePageHeaderContext()` returning just a selected subset of properties, like the counter. But I found it over-killing.